components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -0,0 +1,202 @@
+
+        try {
+            SSLContext sslContext = SSLContexts.custom()
+            //default trust strategy is used (trusting all certificates in the provided trust store).
+                    .loadTrustMaterial(WebSubHubAdapterDataHolder.getInstance().getTrustStore(), null)
+                    .build();
+            builder.setSSLContext(sslContext);
+            builder.setSSLHostnameVerifier(new DefaultHostnameVerifier());
+            LOG.debug("SSL context and hostname verifier added");
+        } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException e) {
+            throw WebSubHubAdapterUtil.handleServerException
+                    (WebSubHubAdapterConstants.ErrorMessages.ERROR_CREATING_SSL_CONTEXT, e);
+        }
+    }
+
+    /**
+     * Create an HTTP POST request.
+     *
+     * @param url The URL for the HTTP POST request.
+     * @param payload The payload to include in the request body.
+     * @return A configured HttpPost instance.
+     * @throws WebSubAdapterException If an error occurs while creating the request.
+     */
+    public HttpPost createHttpPost(String url, Object payload) throws WebSubAdapterException {
+        HttpPost request = new HttpPost(url);
+        request.setHeader(ACCEPT, ContentType.APPLICATION_JSON.getMimeType());
+        request.setHeader(CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType());
+        request.setHeader(CORRELATION_ID_REQUEST_HEADER, WebSubHubAdapterUtil.getCorrelationID());
+
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+
+        try {
+            String jsonString = mapper.writeValueAsString(payload);
+            request.setEntity(new StringEntity(jsonString));
+        } catch (IOException e) {
+            throw WebSubHubAdapterUtil.handleClientException(ERROR_PUBLISHING_EVENT_INVALID_PAYLOAD);
+        }
+
+        return request;
+    }
+
+    /**
+     * Execute an HTTP POST request asynchronously.
+     *
+     * @param httpPost The HTTP POST request to execute.
+     * @return A CompletableFuture containing the HTTP response.
+     */
+    public CompletableFuture<HttpResponse> executeAsync(HttpPost httpPost) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                return httpAsyncClient.execute(httpPost, null).get();
+            } catch (InterruptedException ie) {
+                // Restore interrupted status
+                Thread.currentThread().interrupt();
+                throw new IdentityRuntimeException("Thread was interrupted", ie);
+            } catch (ExecutionException ee) {
+                throw new IdentityRuntimeException("Execution exception", ee);
+            } catch (Exception ex) {
+                throw new IdentityRuntimeException("Exception occurred", ex);
+            }
+        });
+    }
```

Comment:
looks like we are not retrying for failures

Reply:
currently retrying is not incorporated